src/flags/orgFlags.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Flags } from '@oclif/core';
+import { Messages, Org, ConfigAggregator } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
+
+const maybeGetOrg = async (input?: string): Promise<Org | undefined> => Org.create({ aliasOrUsername: input });
+const getOrgOrThrow = async (input?: string): Promise<Org> => {
+  const org = await maybeGetOrg(input);
+  if (!org) {
+    throw messages.createError('errors.NoDefaultEnv');
+  }
+  return org;
+};
+
+const getHubOrThrow = async (aliasOrUsername?: string): Promise<Org> => {
+  if (!aliasOrUsername) {
+    // check config for a default
+    const config = await ConfigAggregator.create();
+    aliasOrUsername = config.getInfo('defaultdevhubusername')?.value as string;
+    if (!aliasOrUsername) {
+      throw messages.createError('errors.NoDefaultHub', [aliasOrUsername]);
+    }
+  }
+  const org = await Org.create({ aliasOrUsername });
+  // check the synchronous, cached version and only use "determine" if we didn't get a positive result
+  if (!org.isDevHubOrg() || !(await org.determineIfDevHubOrg())) {
+    throw messages.createError('errors.NotADevHub', [aliasOrUsername]);
+  }
+  return org;
+};
+
+/**
+ * An optional org specified by username or alias
+ * Will default to the default org if one is not specified.
+ * Will not throw if the specified org and default do not exist
+ *
+ * @example
+ * import { SfCommand, optionalOrgFlag } from '@salesforce/sf-plugins-core';
+ * public static flags = {
+ *     // setting length or prefix
+ *    'target-org': optionalOrgFlag(),
+ *    // adding properties
+ *    'flag2': optionalOrgFlag()({
+ *        required: true,
+ *        description: 'flag2 description',
+ *     }),
+ * }
+ */
+export const optionalOrgFlag = Flags.build<Org | undefined>({
+  char: 'e',
+  parse: async (input: string | undefined) => await maybeGetOrg(input),
+  default: async () => await maybeGetOrg(undefined),
```

Comment:
No need to pass in `undefined` here

Reply:
TS disagrees...how would you do it otherwise?

Reply:
Typescript let's me do it

![Screen Shot 2022-03-01 at 8 51 51 AM](https://user-images.githubusercontent.com/10244328/156201904-92d7c4b8-d452-4657-98c0-5ec08909bbf7.png)